UnityProject/Assets/Scripts/Player/PlayerOnlySyncValues.cs

```diff
@@ -96,6 +173,79 @@ private void SyncAntagState(bool oldState, bool newState)
 			isAntag = newState;
 		}
 
+
+		private void SyncXrayState(bool NotSetValueServer, bool newState)
+		{
+			// fov Occlusion spread = 3 0 if off
+			if ((NotSetValueServer == false && isServer))
+			{
+				SyncXRay = newState;
+			}
+
+			if (ClientForThisBody)
+			{
+				if (newState)
+				{
+					lightingSystem.renderSettings.fovOcclusionSpread = 3;
+				}
+				else
+				{
+					lightingSystem.renderSettings.fovOcclusionSpread = 0;
+				}
+			}
+		}
+
+
+		public void TRaySyncMode(TRayScanner.Mode oldMode, TRayScanner.Mode newMode)
+		{
+			if ((oldMode == TRayScanner.Mode.Off && isServer))
+			{
+				TRayCurrentMode = newMode;
+			}
+
+			if (ClientForThisBody)
+			{
+				var matrixInfos = MatrixManager.Instance.ActiveMatricesList;
+
+				foreach (var matrixInfo in matrixInfos)
+				{
+					var electricalRenderer = matrixInfo.Matrix.ElectricalLayer.GetComponent<TilemapRenderer>();
+					var pipeRenderer = matrixInfo.Matrix.PipeLayer.GetComponent<TilemapRenderer>();
+					var disposalsRenderer = matrixInfo.Matrix.DisposalsLayer.GetComponent<TilemapRenderer>();
+
+					//Turn them all off
+					ChangeState(electricalRenderer, false, 2);
+					ChangeState(pipeRenderer, false, 1);
+					ChangeState(disposalsRenderer, false);
+
+					switch (newMode)
+					{
+						case TRayScanner.Mode.Off:
+							continue;
+						case TRayScanner.Mode.Wires:
+							ChangeState(electricalRenderer, true);
+							continue;
+						case TRayScanner.Mode.Pipes:
+							ChangeState(pipeRenderer, true);
+							continue;
+						case TRayScanner.Mode.Disposals:
+							ChangeState(disposalsRenderer, true);
+							continue;
+						default:
+							Logger.LogError($"Found no case for {newMode}");
+							continue;
+					}
+				}
+			}
+		}
+
+
+		private void ChangeState(TilemapRenderer tileRenderer, bool state, int oldLayerOrder = 0)
+		{
+			tileRenderer.sortingLayerName = state ? "Walls" : "UnderFloor";
+			tileRenderer.sortingOrder = state ? 100 : oldLayerOrder;
+		}
```

Comment:
I feel like this logic can be taken out and put in a static class?

Reply:
can't really think of anywhere to put